draft/carbon-hack-22.pr-faq.md

```diff
@@ -0,0 +1,144 @@
+# Universal Software Carbon Measurement (USCM) API  – PRF
+
+**_USCM is a working title, it will <span style="text-decoration:underline;">not </span>be released under that name._**
+
+_Define Once, Measure Everywhere._
+
+If you can’t measure, you can’t improve. Unfortunately, measuring software carbon emissions is complex and very nuanced; only some people know how to do it well. However, measuring is the biggest bottleneck to reducing the software industry's carbon emissions, so during COP28, the Green Software Foundation announced the launch of USCM. One model to measure everything, everywhere. 
+
+
+## Problems
+
+Modern applications are composed of many smaller pieces of software (components) running on many different _environments_, for example, private cloud, public cloud, bare-metal, virtualized, containerized, mobile, laptops, and desktops.
+
+Every _environment_ requires a different way of measurement, and there is no _single_ solution you can use to calculate the carbon emissions for all environments. 
+
+Measuring requires data. There are free publicly available data sources and higher quality commercial or private data sets. Combining them properly requires a deep understanding of methodologies.
+
+The friction to measuring software emissions isn't that we need to know how, it's that we run software on many _things_ and each _thing_ has several different ways to measure.
+
+There are also two broad categories of measurement, **measurement for reporting** (e.g. GHG) and **measurement for action** (e.g. SCI). Measuring for action requires different measurement methodologies than measuring for reporting.
+
```

Comment:
@jawache With USCM, I believe we are targeting "measurement for action" right?

Reply:
@srini1978 my thoughts are that it should be useful for multiple use cases, e.g. both GHG and SCI (and others). E.g. if you are reporting then it should probably make an opinionated choice and return yearly averages for grid carbon intensity, but if you declare that you are calculating an SCI score it returns real-time grid mix instead. I.e. help users by making some informed decisions for them.

BTW, I was copying comments from Yannick into the doc and i just realized it says measurement for monitoring, I thought it said measurement for reporting. But I think the explanation is the same for both cases, not sure I agree "monitoring"  needs accuracy, or at least when I think of monitoring I think of real-time monitoring of applications rather than a regulatory yearly report.

Reply:
I did indeed mention action and monitoring use cases because the rest of the document let me think we are trying to address both. I agree that real-time monitoring is not monitoring in the sense I defined it (coarse granularity and long period)